File: main/src/main/java/com/sedmelluq/discord/lavaplayer/track/AudioReference.java
package com.sedmelluq.discord.lavaplayer.track;

import com.sedmelluq.discord.lavaplayer.track.info.AudioTrackInfoProvider;

/**
 * An audio item which refers to an unloaded audio item. Source managers can return this to indicate a redirection,
 * which means that the item referred to in it is loaded instead.
 */
public class AudioReference implements AudioItem, AudioTrackInfoProvider {
  public static final AudioReference NO_TRACK = new AudioReference(null, null);

  /**
   * The identifier of the other item.
   */
  public final String identifier;
  /**
   * The title of the other item, if known.
   */
  public final String title;

  /**
   * @param identifier The identifier of the other item.
   * @param title The title of the other item, if known.
   */
  public AudioReference(String identifier, String title) {
    this.identifier = identifier;
    this.title = title;
  }

  @Override
  public String getTitle() {
    return title;
  }

  @Override
  public String getAuthor() {
    return null;
  }

  @Override
  public Long getLength() {
    return null;
  }

  @Override
  public String getIdentifier() {
    return identifier;
  }

  @Override
  public String getUri() {
    return identifier;
  }
}

File: main/src/main/java/lavaplayer/track/AudioReference.kt
package lavaplayer.track

import lavaplayer.container.MediaContainerDescriptor
import lavaplayer.track.info.AudioTrackInfoProvider

/**
 * An audio item which refers to an unloaded audio item. Source managers can return this to indicate a redirection,
 * which means that the item referred to in it is loaded instead.
 *
 * @param identifier The identifier of the other item.
 * @param title      The title of the other item, if known.
 */
class AudioReference @JvmOverloads constructor(
    /**
     * The identifier of the other item.
     */
    @JvmField override val identifier: String?,
    /**
     * The title of the other item, if known.
     */
    @JvmField override val title: String?,
    /**
     * Known probe and probe settings of the item to be loaded.
     */
    @JvmField val containerDescriptor: MediaContainerDescriptor? = null
) : AudioItem, AudioTrackInfoProvider {
    companion object {
        @JvmField
        val NO_TRACK = AudioReference(null, null, null)
    }

    override val author: String?
        get() = null

    override val length: Long?
        get() = null

    override val artworkUrl: String?
        get() = null

    override val uri: String?
        get() = identifier
}
